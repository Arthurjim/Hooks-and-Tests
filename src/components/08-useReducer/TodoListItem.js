import React from "react";
const TodoListItem = ({item,i,handleToogle,deleteTodo}) => {
    return (
        <li className="list-group-item " key={item.id}>
            <p
                className={item.done ? "complete" : ""}
                onClick={() => handleToogle(item.id)}
            >
                {" "}
                {`${i + 1}. ${item.desc}`}
            </p>
            <button
                className="btn btn-danger"
                onClick={() => deleteTodo(item.id)}
            >
                Borrar
            </button>
        </li>
    );
};

export default TodoListItem;
