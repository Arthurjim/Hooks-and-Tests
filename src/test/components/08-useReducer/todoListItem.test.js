import {shallow} from 'enzyme'
import '@testing-library/jest-dom';
import TodoListItem from '../../../components/08-useReducer/TodoListItem'
import { demoTodos,actionAdd,actionRemove, actionToogle} from "../../fixtures/demoTodos";

describe('Pruebas en <todoListItem />', () => {
    const deleteTodo = jest.fn()
    const handleToogle = jest.fn()

    const wrapper = shallow(<TodoListItem key={0} item={demoTodos[0]} i={0} handleToogle={handleToogle} deleteTodo={deleteTodo} />)
    test('Debe de mostrasrse correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe de llamar la funcion handleDelete ', () => {
        wrapper.find('button').simulate('click')
        // expect( deleteTodo ).toHaveBeenCalled();
        expect(deleteTodo).toHaveBeenCalledWith(1)
        //jest.fn()
        //toHaveBeenCalled
        //toHaveBeenCalledWith

    })
      
    test('Debe de llamar la funcion hadleToggle ', () => {
        wrapper.find('p').simulate('click')
        expect(handleToogle).toHaveBeenCalled()

        //jest.fn()
        //toHaveBeenCalledWith
    })
    test('Debe de mostrar el texto correctamente ', () => {
        
        expect(wrapper.find('p').text()).toBe(` 1. ${demoTodos[0].desc}`)
        // {`${i + 1}. ${item.desc}`}
    })
    test('Debe de tener la clase complete ', () => {
        const todo = demoTodos[0];
        todo.done = true
        const wrapper = shallow(<TodoListItem key={0} item={todo} i={0} handleToogle={handleToogle} deleteTodo={deleteTodo} />)

        expect(wrapper.find('p').hasClass('complete')).toBe(true)
        // {`${i + 1}. ${item.desc}`}
    })
})
