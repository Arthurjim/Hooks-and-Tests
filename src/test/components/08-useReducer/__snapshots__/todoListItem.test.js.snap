// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <todoListItem /> Debe de mostrasrse correctamente  1`] = `
<li
  className="list-group-item "
  key="1"
>
  <p
    className=""
    onClick={[Function]}
  >
     
    1. Tareas
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
